activesupport/lib/active_support/duration.rb

```diff
@@ -7,7 +7,15 @@ module ActiveSupport
   #
   #   1.month.ago       # equivalent to Time.now.advance(months: -1)
   class Duration
-    EPOCH = ::Time.utc(2000)
+    PARTS_IN_SECONDS = {
+      seconds:                       1, # Used in parse method for ease of handling part hashes with seconds
+      minutes:                      60,
+      hours:                   60 * 60,
+      days:               24 * 60 * 60,
+      weeks:       7    * 24 * 60 * 60,
+      months:     30    * 24 * 60 * 60,
+      years:    (365.25 * 24 * 60 * 60).ceil,
```

Comment:
It's slightly less clear, but reorganizing this as `1461 * 6 * 60 * 60` (absorbing a factor of 4 from 24 into the first term) will make it an integer like the others.